### PR TITLE
#2508 » Programme page > Overview > Update staff block

### PR DESCRIPTION
**Pull request by engineervix**
[Ticket](https://projects.torchbox.com/projects/rca-django-cms-project/tickets/2508)

This PR makes adjustments to the programme page related staff block as follows:

- remove `max_num=2` limit
- add customisable heading text (with default value of "Staff")
- let related staff be automagically displayed if not manually specified

### Screenshots

<details>
  <summary>When manually specified</summary>

image

</details>

<details>
  <summary>When not specified</summary>

image

</details>

### CODE REVIEW

File: rca/programmes/models.py [hunk 1]
```diff
@@ -819,6 +830,29 @@ def get_programme_stories(self, programme_stories):
...
+        related_staff = self.related_staff.select_related("image")
+        if related_staff:
+            return related_staff
+
+        # For any automatially related staff, adjust the list so we don't have
+        # to make edits to the template shared by other page models.
+        staff = []
+        for item in (
+            StaffPage.objects.filter(roles__programme=self).live().order_by("last_name")
+        ).distinct():
```
Comment (kevinhowbrook) [inline rca/programmes/models.py:852]: Do/Why do we need to call distinct() here?

File: rca/project_styleguide/templates/patterns/organisms/programme_tabs/programme_overview.html [hunk 2]
```diff
@@ -63,7 +63,7 @@ <h3 class="section__heading section__heading--secondary heading heading--four">
 <section class="section section--end bg bg--light">
     {% if related_staff %}
         <div class="section__row section__row--first section__row--last">
-            {% include "patterns/organisms/people-module/people-module.html" with title="Staff" people=related_staff %}
+            {% include "patterns/organisms/people-module/people-module.html" with people=related_staff title=page.staff_title %}
```
Comment (kevinhowbrook) [inline rca/project_styleguide/templates/patterns/organisms/programme_tabs/programme_overview.html:66]: **Suggestion**

As staff_title can be blank, I'd default this to "Staff" if it's blank; unless there is a requirement to have it blank even when staff are listed

File: rca/schools/models.py [hunk 3]
```diff
@@ -424,11 +424,11 @@ class SchoolPage(ContactFieldsMixin, LegacyNewsAndEventsMixin, BasePage):
     staff_panels = [
         FieldPanel("staff_title"),
         FieldPanel("staff_summary"),
-        InlinePanel("related_staff", label="Related staff"),
         HelpPanel(
             content="By default, related staff will be automatically listed. This \
                 can be overriden by adding staff pages here."
         ),
+        InlinePanel("related_staff", label="Related staff"),
```
Comment (kevinhowbrook) [inline rca/schools/models.py:431]: ty for shifting that

**Review by kevinhowbrook — CHANGES_REQUESTED** [review]

File: rca/programmes/models.py (hunk 1 above)
Comment (engineervix) [reply rca/programmes/models.py:852]: When I tested this, I noticed that some staff had multiple roles specified for the same programme. For example:

```console
>>> from rca.people.models import StaffPage
>>> from rca.programmes.models import ProgrammePage
>>> ProgrammePage.objects.live().filter(title='Arts & Humanities')
<PageQuerySet [<ProgrammePage: Arts & Humanities MPhil/PhD>]>
>>> prog = ProgrammePage.objects.live().filter(title='Arts & Humanities').first()
>>> from pprint import pprint as pp
>>> pp(list(StaffPage.objects.filter(roles__programme=prog).live().order_by("last_name")), indent=2)
[ <StaffPage: Professor Gemma Blackshaw>,
  <StaffPage: Dr Nicky Coutts>,
  <StaffPage: Dr Chantal Faust>,
  <StaffPage: Professor Johnny Golding>,
  <StaffPage: Professor Johnny Golding>,
  <StaffPage: Professor Johnny Golding>,
  <StaffPage: Jonathan Miles>,
  <StaffPage: Dr Peter Oakley>]
>>> 
```

Professor Golding appears three times. Here's what the StaffPage looks like in Wagtail admin:

<details>
  <summary>screenshot</summary>

image

</details>

File: rca/project_styleguide/templates/patterns/organisms/programme_tabs/programme_overview.html (hunk 2 above)
Comment (engineervix) [reply rca/project_styleguide/templates/patterns/organisms/programme_tabs/programme_overview.html:66]: In the model, I set the default value of `staff_title` to "Staff":

https://github.com/torchbox/rca-wagtail-2019/blob/b45074da18b8f03a3abc74e44f0982cfc7c7df1c/rca/programmes/models.py#L372-L377

File: rca/programmes/models.py (hunk 1 above)
Comment (kevinhowbrook) [reply rca/programmes/models.py:852]: ah yea that makes a lot of sense! thanks

**Review by kevinhowbrook — APPROVED** [review]

**Pull request merged**